docs/core/knowledge-bases.rst

```diff
@@ -0,0 +1,556 @@
+in different ways.
+Our action is able to (1) resolve ordinal mentions, such as "the first one", to the actual object and (2) resolve
+mentions, such as "it" or "that one", to the last mentioned object in the conversation.
+Let's take a closer look.
+
+**Ordinal Mentions**
+
+If the user refers to an object by its position in a list, we talk about ordinal mentions.
+Let's look at an example conversation:
+
+- User: `What restaurants in Berlin do you know?`
+- Bot: `Found the following objects of type 'restaurant':  1: I due forni  2: PastaBar  3: Berlin Burrito Company`
+- User: `Does the first one have outside seating?`
+
+The user referred to "I due forni" by the term "the first one".
+Other ordinal mentions are, for example:
+
+- `the second one`
+- `the last one`
+- `any`
+- `3`
+
+Ordinal mentions are typically used when a list of objects was presented to the user.
+To resolve those mentions to the actual object, we use an ordinal mention mapping which is set in the
+``KnowledgeBase`` class.
+The default mapping looks like the following:
+
+  .. code-block:: python
+
+      {
+          "1": lambda l: l[0],
+          "2": lambda l: l[1],
+          "3": lambda l: l[2],
+          "4": lambda l: l[3],
+          "5": lambda l: l[4],
+          "6": lambda l: l[5],
+          "7": lambda l: l[6],
+          "8": lambda l: l[7],
+          "9": lambda l: l[8],
+          "10": lambda l: l[9],
+          "ANY": lambda l: random.choice(list),
+          "LAST": lambda l: l[-1],
+      }
+
+The ordinal mention mapping maps a string, such as "1", to the object in a list, e.g. ``lambda l: l[0]``.
+You can overwrite the ordinal mention mapping by calling the function ``set_ordinal_mention_mapping()`` on your
+``KnowledgeBase`` implementation (see :ref:`customize_in_memory_knowledge_base`).
+As the ordinal mention mapping does not, for example, include an entry for "the first one".
+It is important that you use :ref:`entity_synonyms` to map "the first one" in your NLU data to "1".
+For example,
+
+.. code-block:: md
+
+    Does the [first one](mention:1) have [outside seating](attribute:outside-seating)?
+
+maps "first one" via a synonym to "1".
+The NER detects "first one" as ``mention`` entity, but puts "1" into the ``mention`` slot.
+Thus, our action can take the ``mention`` slot together with the ordinal mention mapping to resolve "first one" to
+the actual object "I due forni".
+
+Other Mentions
+~~~~~~~~~~~~~~
+Take a look at the following conversation:
+
+- User: `What is the cuisine of PastaBar?`
+- Bot: `PastaBar has an Italian cuisine.`
+- User: `Does it have wifi?`
+- Bot: `Yes.`
+- User: `Can you give me an address?`
+
+In the second utterance of the user, the user refers to "PastaBar" by the word "it".
+If the NER detected "it" as the entity ``mention``, the knowledge base action would resolve it to the last mentioned
+object in the conversation, e.g. "PastaBar".
```

Comment:
What do you mean? The NER should detect "it" as `mention`. Otherwise the action might not be able to resolve it to the last mentioned object (in case `use_last_object_mention` is set to False).

Reply:
Ah okay. I just mean when creating the NLU data, then we should also be tagging `it` as a mention? e.g. in L160:
```
what [cuisine](attribute) is it?
```
we don't tag `it` as a mention

Reply:
good catch! will add it.

Reply:
makes sense! another couple of examples of mentions like `it` and `that one` would be helpful